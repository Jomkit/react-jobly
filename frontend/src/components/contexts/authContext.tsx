import { createContext } from 'react';
import { authInterface } from '../../types';

export const authContext = createContext<authInterface>({} as authInterface);