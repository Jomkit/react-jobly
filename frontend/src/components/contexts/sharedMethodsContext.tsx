import { createContext } from 'react';
import { sharedMethodsInterface } from '../../types';

export const sharedMethodsContext = createContext<sharedMethodsInterface>({} as sharedMethodsInterface);