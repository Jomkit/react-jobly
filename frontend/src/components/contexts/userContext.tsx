import { createContext } from "react";
import { userInterface } from "../../types";

export const userContext = createContext<userInterface | null>(null);