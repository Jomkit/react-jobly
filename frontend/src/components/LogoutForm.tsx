import { FormEvent, useContext } from 'react'
import { authInterface } from '../types';
import { authContext } from './contexts/authContext';

const LogoutForm = () => {
    const { logout } = useContext<authInterface>(authContext);

    const handleSubmit = (evt: FormEvent) => {
        evt.preventDefault();
        logout();
    }

  return (
    <form method='post' onSubmit={handleSubmit}>
        <button className='btn text-light' type="submit"><strong>Logout</strong></button>
    </form>
  )
}

export default LogoutForm