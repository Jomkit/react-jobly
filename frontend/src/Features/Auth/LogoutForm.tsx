import { FormEvent, useContext } from 'react'
import { sharedMethodsInterface } from '/home/jomkit/react-jobly/frontend/src/types.ts';
import { sharedMethodsContext } from '../../components/contexts/sharedMethodsContext';

const LogoutForm = () => {
    const { logout } = useContext<sharedMethodsInterface>(sharedMethodsContext);

    const handleSubmit = (evt: FormEvent) => {
        evt.preventDefault();
        logout();
    }

  return (
    <form method='post' onSubmit={handleSubmit}>
        <button className='btn text-light' type="submit"><strong>Logout</strong></button>
    </form>
  )
}

export default LogoutForm