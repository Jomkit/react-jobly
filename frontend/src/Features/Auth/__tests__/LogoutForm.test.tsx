import userEvent from "@testing-library/user-event";
import { MemoryRouter } from "react-router-dom";
import { sharedMethodsContext } from "../../../components/contexts/sharedMethodsContext";
import { render } from "@testing-library/react";
import LogoutForm from "../LogoutForm";

function setup(tsx: React.ReactNode) {
    return {
        user: userEvent.setup(),
        ...render(tsx)
    }
}

const signup = vi.fn();
const login = vi.fn();
const logout = vi.fn();

const TestLogoutForm = () => {
    return (
        <MemoryRouter initialEntries={["/"]}>
            <sharedMethodsContext.Provider value={{signup, login, logout}} >
                <LogoutForm />
            </sharedMethodsContext.Provider>
        </MemoryRouter>
    )
}

it('should should render without crashing', () => { 
    render(<TestLogoutForm />);
})

it('should match snapshot', () => { 
    const { asFragment } = render(<TestLogoutForm />);
    expect(asFragment()).toMatchSnapshot();
})

it('should call logout function when logout button clicked', async () => { 
    const { user, getByText } = setup(<TestLogoutForm/>);

    const logoutBtn = getByText("Logout");

    await user.click(logoutBtn);
    expect(logout).toHaveBeenCalled();
})
