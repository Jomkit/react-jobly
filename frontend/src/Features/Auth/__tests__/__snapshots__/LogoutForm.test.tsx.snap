// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form
    method="post"
  >
    <button
      class="btn text-light"
      type="submit"
    >
      <strong>
        Logout
      </strong>
    </button>
  </form>
</DocumentFragment>
`;
